### PR TITLE
Add a separate ceiling for each itemized deduction

### PR DESCRIPTION
**Pull request by hdoupe**
This pull request addresses issue #1384.  New tax law parameters were created to give one the option to place a separate ceiling on each itemized deduction parameter.

cc @MattHJensen

### CODE REVIEW

**Comment by martinholmer** [comment]
@hdoupe,  Thanks for your contribution in pull request #1385.  After a quick review it looks good to me, but these enhancements were requested by @MattHJensen.  So, I think it is best if he gives #1385 a review before merging it into master.  @MattHJensen, how does #1385 look to you?

**Comment by MattHJensen** [comment]
The implementation looks good. Thanks @hdoupe 

Could you move each ceiling parameter into the same section of the file as the corresponding haircut and give it the same section attributes. For example, the casualty ceiling would have:

        "section_1": "Itemized Deductions",
        "section_2": "Casualty",

**Comment by hdoupe** [comment]
@MattHJensen @martinholmer Thanks for reviewing my pull request.  While making the suggested changes, I realized that I forgot to add a ceiling for the real estate tax exemption.  The most recent PR contains this addition.

I'll fix the pep8 issues in the morning.

**Comment by MattHJensen** [comment]
@hdoupe, as a simple test of this PR, could you also update the renacci.json to use the new ceilings parameters and make sure you get the same result? If so, you can include the renacci.json commit in this PR.

**Comment by MattHJensen** [comment]
@hdoupe, could you also add an entry to RELEASES.md in this PR

**Comment by hdoupe** [comment]
@MattHJensen I replaced _ID_c with _ID_InterestPaid_c and the number of itemizers increased by about 1 million and the itemized deductions increased by $65 billion in 2017 to $96.4 billion in 2024.

Perhaps, the difference arises from _ID_c applying to both charity and interest paid deductions, since there is no hair cut on the charity deduction.  I tried to isolate this by setting _ID_Charity_hc to 1 in the first reform and _ID_Charity_c to 0 in the second reform and the results were more similar, but the number of itemizers decreased by about 100,000 and the itemized deductions decreased by $1.1 billion in 2017 to $1.9 billion in 2026.

I'm going to look into this more to see why there is a difference between the reform using the hair cuts and total ceiling (ID_c) and the reform using the separate ceilings.

**Comment by hdoupe** [comment]
@MattHJensen Here's why the numbers were different.  ID_c is applied after the gross itemized deduction is calculated with the phaseout, floor, etc.  ID_InterestPaid_c and the other separate ceilings are applied before this adjustment is made.  Also, without the separate ceilings, Renacci's ceiling on the interest paid was applied to both the interest paid and the charity expense.  

After accounting for these two issues, the results for the reform using ID_c and the results for the reform using a separate ceiling for interest paid were the same.

**Comment by hdoupe** [comment]
Close #1384

File: RELEASES.md [hunk 1]
```diff
@@ -19,9 +19,13 @@ Release 0.8.5 on 2017-06-??
...
-  and
-  [#1383](https://github.com/open-source-economics/Tax-Calculator/pull/1383)
-  by Hank Doupe]
+  ,[#1383](https://github.com/open-source-economics/Tax-Calculator/pull/1383)
+  , and
+  [#1385](https://github.com/open-source-economics/Tax-Calculator/pull/1385)
+  by Hank Doupe
+- Add separate ceiling for each itemized deduction parameter
+  [#1385](https://github.com/open-source-economics/Tax-Calculator/pull/1385)
+  by Hank Doupe
```
Comment (MattHJensen) [inline RELEASES.md]: Check your double brackets here. You can view how the file is rendering 
image

File: RELEASES.md [hunk 2]
```diff
@@ -463,7 +467,7 @@ Release 0.6.6 on 2016-08-13
 **API Changes**
 - None
 
-**New Features** 
+**New Features**
```
Comment (MattHJensen) [inline RELEASES.md]: Thanks for the little improvements too!

**Comment by MattHJensen** [comment]
> Here's why the numbers were different. ID_c is applied after the gross itemized deduction is calculated with the phaseout, floor, etc. ID_InterestPaid_c and the other separate ceilings are applied before this adjustment is made. Also, without the separate ceilings, Renacci's ceiling on the interest paid was applied to both the interest paid and the charity expense.

> After accounting for these two issues, the results for the reform using ID_c and the results for the reform using a separate ceiling for interest paid were the same.

Got it. Thank you. I will merge as soon as you fix the bracket formatting in RELEASES.md.

**Comment by hdoupe** [comment]
@MattHJensen Thanks for the review.  I believe the small changes were made by my text editor.

**Comment by MattHJensen** [comment]
> I believe the small changes were made by my text editor.

Nice feature. What do you use?

**Comment by hdoupe** [comment]
I use [Atom](https://atom.io/). @andersonfrailey showed it to me.  It's a github project and it's open source.  I like it so far.  It has some pretty convenient features such as highlighting files that you've edited after you open a new branch.

**Comment by MattHJensen** [comment]
@hdoupe, it looks like the double brackets are still a problem. Only the first entry here has them right. 
image

**Comment by hdoupe** [comment]
@MattHJensen Now the brackets look good.  Sorry about that.

**Comment by martinholmer** [comment]
@hdoupe,  Be sure to update your master branch and then merge it into your `add_deduction_ceilings` branch in order to incorporate recent changes I made to `RELEASES.md`.

**Comment by hdoupe** [comment]
@martinholmer Ok, I just took care of that.  Should I do another pull request for the RELEASES.md after the merge?

**Comment by hdoupe** [comment]
@martinholmer The RELEASES.md is up to date now.  This should be ready to merge.

cc @MattHJensen

**Comment by martinholmer** [comment]
@hdoupe,  Thanks for the help on the `RELEASES.md` document in pull request #1385.
I'm going to defer to @MattHJensen on the final review and merge into the master branch.

**Pull request merged**